projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -74,6 +80,12 @@ async function getIssueReferences( octokit, owner, repo, number, issueComments )
 	ticketReferences.map( reference => {
 		const supportId = reference[ 0 ];
 
+		// Do not correct p2 comment references.
+		if ( supportId.includes( '-p2#comment-' ) ) {
+			correctedSupportIds.add( supportId );
+			return;
+		}
+
 		// xxx-zen is the preferred format for tickets.
 		// xxx-zd, as well as its uppercase version, is considered an alternate version.
 		const wrongId = supportId.match( /([0-9]*)-zd/i );
```

Comment:
We should probably anchor this pattern
```suggestion
		const wrongId = supportId.match( /^([0-9]*)-zd$/i );
```
After doing that, the check being added above shouldn't be needed since there's no way this could match.